docs/enhancements/infrastructure-monitoring-service-api/alarms.md

```diff
@@ -0,0 +1,669 @@
+
+## Ready
+
+### For a given ResourceTypeID and AlarmName (coming from AM alert), find the AlarmDefinitionID and ProbableCauseID
+- Find resourceType ID and alert name from current Alertmanager payload
+  - Get managedcluster-id and alert name from alertmanager alerts (e.g NodeClockNotSynchronising2)
+  - Ask inventory for ResourceType ID using managedcluster-id (e.g b3e7149e-d471-4d0f-aaa6-d5e9aa9e713a)
+    ```go
+    // should be something we compute 
+    func getResourceTypeID(managedClusterId uuid.UUID, class ResourceClass, resourceType ResourceType) uuid.UUID {
+      return v5UUID(fmt.Sprintf("%s-%s-%s", managedClusterId.String(), class, resourceType))
+    }
+    ```
+- Run following. `resource_type_id` should be enough to uniquely identify the corresponding alarm dictionary
+  and its associated alarm definitions and finally use `alarm_name` to filter out the exact rule.
+    ```sql
+    SELECT
+        ad.alarm_definition_id,
+        pc.probable_cause_id
+    FROM
+        alarm_definitions ad
+        JOIN
+          alarm_dictionary adict
+          ON ad.alarm_dictionary_version = adict.alarm_dictionary_version
+        JOIN
+          probable_causes pc
+          ON ad.alarm_definition_id = pc.alarm_definition_id
+    WHERE
+        adict.resource_type_id = '481688c8-2782-4534-a9de-88ca5154411d'
+        AND ad.alarm_name = 'LowMemory';
+    ```
+
+
+### Notification tracking
+- Collect all subscription info including ID, callback and filter
+- For each subscription, collect all the `AlarmEventRecord` rows based on sequence number and optionally the filter. Here we are collecting everything that's "CRITICAL"
+    ```sql
+    SELECT aer.*
+    FROM alarm_event_record aer
+             JOIN alarm_subscription_info asi ON asi.alarm_subscription_id = 'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11'
+    WHERE aer.alarm_sequence_number > asi.largest_number_alarm_event_seen_so_far
+      and aer.perceived_severity = 'CRITICAL'
+    ORDER BY aer.alarm_sequence_number;
+    ```
+- Process and notify by deriving `AlarmEventRecordModifications` o-ran DS + callback.
+- Update sequence for subscription indicating the latest event sent so far
+    ```go
+    var largestProcessedSequenceNumber int64
+    // for each alarms Update the largest sequence number we've processed
+    largestProcessedSequenceNumber = max(alarm.SequenceNumber, largestProcessedSequenceNumber))
+    // The last row in the query is always the largest (ORDER BY default is accending)
+    ```
+  And finally update 
+  ```sql
+  UPDATE alarm_subscription_info
+  SET largest_number_alarm_event_seen_so_far = $largestProcessedSequenceNumber
+  WHERE alarm_subscription_id = 'a2eebc99-9c0b-4ef8-bb6d-6bb9bd380a13';
+  ```
+- NOTE: `alarm_sequence_number` is automatically handled from inside the DB. When the sequence increments, subscriber is notified. 
+  See notification conditions [here](#conditions-for-notifying-subscriber)
+  
+### Conditions for Notifying subscriber
```

Comment:
Is this something O-RAN specifies? If so a reference would be good.

Reply:
No actually we dont have anything specific...but in the doc it says 
```
The Alarm Change Notification is used by the O-Cloud Infrastructure Monitoring Service to report changes to the O2ims alarm List to an O2ims consumer that has subscribed to such notifications.
``` 
So a change it our case it basically firing for the first time, resolved and when/if the alarmChangedTime updates.  I'll update the doc to include this info

Reply:
Refer to 3.7.2 Alarm Notification Use Case in O-RAN-WG6.ORCH-USE-CASES-R003-v10.00

Reply:
updated and also included the download link